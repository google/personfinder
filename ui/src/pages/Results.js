/*
 * Copyright 2019 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React, {Component} from 'react';
import {FormattedDate, FormattedMessage, defineMessages, injectIntl} from 'react-intl';
import {withRouter} from 'react-router-dom';
import Fab from '@material/react-fab';

import Footer from './../components/Footer.js';
import LoadingIndicator from './../components/LoadingIndicator.js';
import RepoHeader from './../components/RepoHeader.js';
import SearchBar from './../components/SearchBar.js';
import Utils from './../utils/Utils.js';

const MESSAGES = defineMessages({
  createNewRecord: {
    id: 'Results.createNewRecord',
    defaultMessage: 'Create new record',
    description: 'Label on a button for creating a new database entry.',
  },
  noResultsFound: {
    id: 'Results.noResultsFound',
    defaultMessage: 'No results found.',
    description: 'A message shown when a search query has produced no results.',
  },
  timestampCreation: {
    id: 'Results.timestampCreation',
    defaultMessage: 'Record created on {timestampStr}.',
    description: 'A message saying when a record was originally created.',
  },
  timestampUpdate: {
    id: 'Results.timestampUpdate',
    defaultMessage: 'Record updated on {timestampStr}.',
    description: 'A message saying when a record was last updated.',
  },
});

/*
 * A component for showing a search results page.
 */
class Results extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isLoaded: false,
      error: null,
      repo: null,
      results: null
    };
    this.repoId = this.props.match.params.repoId;
    this.goToCreate = this.goToCreate.bind(this);
    this.loadResults = this.loadResults.bind(this);
    this.handleSearch = this.handleSearch.bind(this);
  }

  goToCreate() {
    this.props.history.push({
          pathname: `/${this.repoId}/create`,
      });
  }

  loadResults() {
    const query = Utils.getURLParam(this.props, 'query_name');
    const apiURLs = [
        `/${this.repoId}/d/repo`,
        '/' + this.repoId + '/d/results?query=' + encodeURIComponent(query),
        ];
    Promise.all(apiURLs.map(url => fetch(url)))
        .then(res => Promise.all(res.map(r => r.json())))
        .then(
          ([repo, results]) => {
            this.setState({
              isLoaded: true,
              repo: repo,
              results: results,
            });
          },
          (error) => {
            this.setState({error: error});
          }
        );
  }

  componentDidMount() {
    this.loadResults();
  }

  componentDidUpdate(prevProps) {
    if (this.props != prevProps) {
      this.loadResults();
    }
  }

  handleSearch(query) {
    this.setState({isLoaded: false});
    this.props.history.push({
        pathname: `/${this.repoId}/results`,
        search: '?query_name=' + encodeURIComponent(query),
      });
  }

  renderResults() {
    if (this.state.results.length == 0) {
      return (
        <p className='mdc-typography--body1'>
          <FormattedMessage {...MESSAGES.noResultsFound} />
        </p>
      );
    }
    const results = this.state.results.map(result => (
      <SearchResult
        key={result.personId}
        repo={this.state.repo}
        result={result}
      />
    ));
    return <ul className='results-list'>{results}</ul>;
  }

  renderAddPersonFab() {
    return (
      <Fab
          className='results-addfab'
          onClick={this.goToCreate}
          icon={<img src='/static/icons/maticon_add.svg' />}
          textLabel={this.props.intl.formatMessage(
              MESSAGES.createNewRecord)}
      />
    );
  }

  render() {
    if (this.state.error) {
      return <div>An error occurred</div>
    }
    if (!this.state.isLoaded) {
      return <LoadingIndicator />;
    }
    return (
      <div id='results-wrapper'>
        <RepoHeader
          repo={this.state.repo}
          backButtonTarget={`/${this.repoId}`}
        />
        <div className='results-body'>
          <SearchBar
            repoId={this.repoId}
            initialValue={Utils.getURLParam(this.props, 'query_name')}
            onSearch={this.handleSearch}
          />
          {this.renderResults()}
          <Footer />
        </div>
        {this.renderAddPersonFab()}
      </div>
    );
  }
}

/**
 * A component for an individual search result.
 */
class SearchResultImpl extends Component {
  constructor(props) {
    super(props);
    this.goToView = this.goToView.bind(this);
  }

  goToView() {
    this.props.history.push('/' + this.props.repo.repoId + '/view?id='
        + encodeURIComponent(this.props.result.personId));
  }

  render() {
    var nameStr = this.props.result.fullNames.join(', ');
    if (this.props.result.alternateNames.length > 0) {
      nameStr += ' (' + this.props.result.alternateNames.join(', ') + ')';
    }
    const formattedTimestamp = <FormattedDate
        value={new Date(this.props.result.timestamp)}
        day='numeric'
        month='short'
        hour='numeric'
        minute='numeric'
        timeZoneName='short' />
    var timestampLine = '';
    if (this.props.result.timestampType == 'creation') {
      timestampLine = (
          <FormattedMessage
            {...MESSAGES.timestampCreation}
            values={{
              'timestampStr': formattedTimestamp,
            }} />);
    } else if (this.props.result.timestampType == 'update') {
      timestampLine = (
          <FormattedMessage
            {...MESSAGES.timestampUpdate}
            values={{
              'timestampStr': formattedTimestamp,
            }} />);
    }
    return (
      <li className='results-result' onClick={this.goToView}>
        <div className='results-resultphoto'>
          <img src={this.props.result.localPhotoUrl} />
        </div>
        <div className='results-resultcontent'>
          <h5 className='mdc-typography--headline5'>
            {nameStr}
          </h5>
          <p className='mdc-typography--body1'>
            {timestampLine}
          </p>
        </div>
      </li>
    );
  }
}

const SearchResult = withRouter(injectIntl(SearchResultImpl));

export default injectIntl(Results);
