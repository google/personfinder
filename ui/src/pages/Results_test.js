/*
 * Copyright 2019 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import {createMemoryHistory} from 'history';
import React from 'react';
import {MemoryRouter} from 'react-router';

import Results from './Results';
import RepoHeader from '../components/RepoHeader';
import SearchBar from '../components/SearchBar';
import {mountWithIntl} from '../testing/enzyme-intl-helper';
import {flushPromises} from '../testing/utils';

Enzyme.configure({adapter: new Adapter()});

const REPO_DATA = {repoId: 'albany', title: 'Albany', recordCount: 100,};
const RESULTS_DATA = [
    {
      personId: '123',
      fullNames: ['Fred Fredricks'],
      alternateNames: ['Freddy'],
      timestampType: 'creation',
      timestamp: '2019-05-15T17:12:23.936282Z',
      localPhotoUrl: null,
    },
    {
      personId: '321',
      fullNames: ['Alan Smith', 'Alan Herbert Smith'],
      alternateNames: [],
      timestampType: 'update',
      timestamp: '2019-05-16T16:35:23.936282Z',
      localPhotoUrl: 'http://www.example.com/notevenreallylocal.jpg',
    },
  ]

function setupPageWrapper() {
  fetch.mockResponseOnce(JSON.stringify(REPO_DATA));
  fetch.mockResponseOnce(JSON.stringify(RESULTS_DATA));
  const history = createMemoryHistory('/albany');
  const locationValue = {search: 'query_name=th%C3%A1tcher'};
  const matchValue = {params: {repoId: 'albany'}};
  const wrapper = mountWithIntl(
    <MemoryRouter>
      <Results
          history={history}
          location={locationValue}
          match={matchValue} />
    </MemoryRouter>
  );
  return [wrapper, history];
}

describe('testing Results', () => {
  beforeEach(() => {
    fetch.resetMocks();
  });

  test('Results calls to correct API URLs', () => {
    const [wrapper] = setupPageWrapper();
    return flushPromises().then(() => {
      wrapper.update();
      expect(fetch.mock.calls.length).toBe(2);
      expect(fetch.mock.calls[0][0]).toBe('/albany/d/repo');
      expect(fetch.mock.calls[1][0]).toBe(
          '/albany/d/results?query=th%C3%A1tcher');
      wrapper.unmount();
    });
  });

  test('RepoHeader configured correctly', () => {
    const [wrapper] = setupPageWrapper();
    return flushPromises().then(() => {
      wrapper.update();
      const actualRepoHeader = wrapper.find(RepoHeader).get(0);
      expect(actualRepoHeader.props.repo).toEqual(REPO_DATA);
      expect(actualRepoHeader.props.backButtonTarget).toBe('/albany');
      wrapper.unmount();
    });
  });

  test('SearchBar configured correctly', () => {
    const [wrapper, history] = setupPageWrapper();
    return flushPromises().then(() => {
      wrapper.update();
      const actualSearchBar = wrapper.find(SearchBar).get(0);
      expect(actualSearchBar.props.repoId).toBe('albany');
      expect(actualSearchBar.props.initialValue).toBe('thátcher');
      actualSearchBar.props.onSearch('fródo');
      expect(history.entries[1].pathname).toBe('/albany/results');
      expect(history.entries[1].search).toBe('?query_name=fr%C3%B3do');
      wrapper.unmount();
    });
  });

  test('"Add Person" FAB points to create page', () => {
    const [wrapper, history] = setupPageWrapper();
    return flushPromises().then(() => {
      wrapper.update();
      wrapper.find('.results-addfab').at(0).simulate('click');
      expect(history.entries[1].pathname).toBe('/albany/create');
      wrapper.unmount();
    });
  });

  test('snapshot test for Results', () => {
    // We don't use setupPageWrapper here because we want to avoid passing a
    // history object and need to specify initialEntries, to avoid generating
    // random keys that mess up the snapshot.
    fetch.mockResponseOnce(JSON.stringify(REPO_DATA));
    fetch.mockResponseOnce(JSON.stringify(RESULTS_DATA));
    const locationValue = {search: 'query_name=th%C3%A1tcher'};
    const matchValue = {params: {repoId: 'albany'}};
    const wrapper = mountWithIntl(
      <MemoryRouter
          initialEntries={[{pathname: '/albany/results', key: 'abc123'}]}>
        <Results
            location={locationValue}
            match={matchValue} />
      </MemoryRouter>
    );
    return flushPromises().then(() => {
      wrapper.update();
      expect(toJson(wrapper.find(Results).at(0))).toMatchSnapshot();
      wrapper.unmount();
    });
  });
});
