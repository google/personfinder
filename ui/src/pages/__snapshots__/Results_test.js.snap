// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Results snapshot test for Results 1`] = `
<InjectIntl(Results)
  location={
    Object {
      "search": "query_name=th%C3%A1tcher",
    }
  }
  match={
    Object {
      "params": Object {
        "repoId": "albany",
      },
    }
  }
>
  <Results
    intl={
      Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "en",
        "messages": Object {},
        "now": [Function],
        "onError": [Function],
        "textComponent": "span",
        "timeZone": null,
      }
    }
    location={
      Object {
        "search": "query_name=th%C3%A1tcher",
      }
    }
    match={
      Object {
        "params": Object {
          "repoId": "albany",
        },
      }
    }
  >
    <div
      id="results-wrapper"
    >
      <InjectIntl(RepoHeader)
        backButtonTarget="/albany"
        repo={
          Object {
            "recordCount": 100,
            "repoId": "albany",
            "title": "Albany",
          }
        }
      >
        <RepoHeader
          backButtonTarget="/albany"
          intl={
            Object {
              "defaultFormats": Object {},
              "defaultLocale": "en",
              "formatDate": [Function],
              "formatHTMLMessage": [Function],
              "formatMessage": [Function],
              "formatNumber": [Function],
              "formatPlural": [Function],
              "formatRelative": [Function],
              "formatTime": [Function],
              "formats": Object {},
              "formatters": Object {
                "getDateTimeFormat": [Function],
                "getMessageFormat": [Function],
                "getNumberFormat": [Function],
                "getPluralFormat": [Function],
                "getRelativeFormat": [Function],
              },
              "locale": "en",
              "messages": Object {},
              "now": [Function],
              "onError": [Function],
              "textComponent": "span",
              "timeZone": null,
            }
          }
          repo={
            Object {
              "recordCount": 100,
              "repoId": "albany",
              "title": "Albany",
            }
          }
        >
          <div
            id="repoheader"
          >
            <div
              id="repoheader-backbutton"
            >
              <Link
                replace={false}
                to="/albany"
              >
                <a
                  href="/albany"
                  onClick={[Function]}
                >
                  <div>
                    <img
                      src="/static/icons/maticon_arrow_back.svg"
                    />
                  </div>
                </a>
              </Link>
            </div>
            <div
              id="repoheader-info"
            >
              <p
                className="mdc-typography--subtitle1"
              >
                <FormattedMessage
                  defaultMessage="Google Person Finder"
                  id="RepoHeader.productName"
                  values={Object {}}
                >
                  <span>
                    Google Person Finder
                  </span>
                </FormattedMessage>
              </p>
              <p
                className="mdc-typography--subtitle2"
              >
                Albany
              </p>
            </div>
          </div>
        </RepoHeader>
      </InjectIntl(RepoHeader)>
      <div
        className="results-body"
      >
        <InjectIntl(SearchBar)
          initialValue="th치tcher"
          onSearch={[Function]}
          repoId="albany"
        >
          <SearchBar
            initialValue="th치tcher"
            intl={
              Object {
                "defaultFormats": Object {},
                "defaultLocale": "en",
                "formatDate": [Function],
                "formatHTMLMessage": [Function],
                "formatMessage": [Function],
                "formatNumber": [Function],
                "formatPlural": [Function],
                "formatRelative": [Function],
                "formatTime": [Function],
                "formats": Object {},
                "formatters": Object {
                  "getDateTimeFormat": [Function],
                  "getMessageFormat": [Function],
                  "getNumberFormat": [Function],
                  "getPluralFormat": [Function],
                  "getRelativeFormat": [Function],
                },
                "locale": "en",
                "messages": Object {},
                "now": [Function],
                "onError": [Function],
                "textComponent": "span",
                "timeZone": null,
              }
            }
            onSearch={[Function]}
            repoId="albany"
          >
            <TextField
              className="searchbar"
              dense={false}
              floatingLabelClassName=""
              fullWidth={false}
              label="Search for a person"
              lineRippleClassName=""
              notchedOutlineClassName=""
              outlined={true}
              textarea={false}
            >
              <div
                className="mdc-text-field searchbar mdc-text-field--outlined"
                key="text-field-container"
                onClick={[Function]}
                onKeyDown={[Function]}
              >
                <Input
                  className=""
                  disabled={false}
                  foundation={
                    MDCTextFieldFoundation {
                      "adapter_": Object {
                        "activateLineRipple": [Function],
                        "addClass": [Function],
                        "closeOutline": [Function],
                        "deactivateLineRipple": [Function],
                        "deregisterInputInteractionHandler": [Function],
                        "deregisterTextFieldInteractionHandler": [Function],
                        "deregisterValidationAttributeChangeHandler": [Function],
                        "floatLabel": [Function],
                        "getLabelWidth": [Function],
                        "getNativeInput": [Function],
                        "hasClass": [Function],
                        "hasLabel": [Function],
                        "hasOutline": [Function],
                        "isFocused": [Function],
                        "notchOutline": [Function],
                        "registerInputInteractionHandler": [Function],
                        "registerTextFieldInteractionHandler": [Function],
                        "registerValidationAttributeChangeHandler": [Function],
                        "removeClass": [Function],
                        "setLineRippleTransformOrigin": [Function],
                        "shakeLabel": [Function],
                      },
                      "characterCounter_": undefined,
                      "helperText_": undefined,
                      "inputBlurHandler_": [Function],
                      "inputFocusHandler_": [Function],
                      "inputInputHandler_": [Function],
                      "isFocused_": false,
                      "isValid_": true,
                      "leadingIcon_": undefined,
                      "receivedUserInput_": false,
                      "setPointerXOffset_": [Function],
                      "textFieldInteractionHandler_": [Function],
                      "trailingIcon_": undefined,
                      "useNativeValidation_": true,
                      "validationAttributeChangeHandler_": [Function],
                      "validationObserver_": undefined,
                    }
                  }
                  handleFocusChange={[Function]}
                  handleValueChange={[Function]}
                  id=""
                  inputType="input"
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  setDisabled={[Function]}
                  setInputId={[Function]}
                  syncInput={[Function]}
                  value="th치tcher"
                >
                  <input
                    className="mdc-text-field__input"
                    disabled={false}
                    id=""
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onMouseDown={[Function]}
                    onTouchStart={[Function]}
                    value="th치tcher"
                  />
                </Input>
                <NotchedOutline
                  className=""
                  notch={true}
                  notchWidth={0}
                >
                  <div
                    className="mdc-notched-outline mdc-notched-outline--upgraded mdc-notched-outline--notched"
                  >
                    <div
                      className="mdc-notched-outline__leading"
                    />
                    <div
                      className="mdc-notched-outline__notch"
                      style={
                        Object {
                          "width": undefined,
                        }
                      }
                    >
                      <FloatingLabel
                        className=""
                        float={true}
                        handleWidthChange={[Function]}
                        htmlFor=""
                      >
                        <label
                          className="mdc-floating-label mdc-floating-label--float-above"
                          htmlFor=""
                          onAnimationEnd={[Function]}
                        >
                          Search for a person
                        </label>
                      </FloatingLabel>
                    </div>
                    <div
                      className="mdc-notched-outline__trailing"
                    />
                  </div>
                </NotchedOutline>
              </div>
            </TextField>
          </SearchBar>
        </InjectIntl(SearchBar)>
        <ul
          className="results-list"
        >
          <withRouter(InjectIntl(SearchResultImpl))
            key="123"
            repo={
              Object {
                "recordCount": 100,
                "repoId": "albany",
                "title": "Albany",
              }
            }
            result={
              Object {
                "alternateNames": Array [
                  "Freddy",
                ],
                "fullNames": Array [
                  "Fred Fredricks",
                ],
                "localPhotoUrl": null,
                "personId": "123",
                "timestamp": "2019-05-15T17:12:23.936282Z",
                "timestampType": "creation",
              }
            }
          >
            <Route>
              <InjectIntl(SearchResultImpl)
                history={
                  Object {
                    "action": "POP",
                    "block": [Function],
                    "canGo": [Function],
                    "createHref": [Function],
                    "entries": Array [
                      Object {
                        "hash": "",
                        "key": "abc123",
                        "pathname": "/albany/results",
                        "search": "",
                      },
                    ],
                    "go": [Function],
                    "goBack": [Function],
                    "goForward": [Function],
                    "index": 0,
                    "length": 1,
                    "listen": [Function],
                    "location": Object {
                      "hash": "",
                      "key": "abc123",
                      "pathname": "/albany/results",
                      "search": "",
                    },
                    "push": [Function],
                    "replace": [Function],
                  }
                }
                location={
                  Object {
                    "hash": "",
                    "key": "abc123",
                    "pathname": "/albany/results",
                    "search": "",
                  }
                }
                match={
                  Object {
                    "isExact": false,
                    "params": Object {},
                    "path": "/",
                    "url": "/",
                  }
                }
                repo={
                  Object {
                    "recordCount": 100,
                    "repoId": "albany",
                    "title": "Albany",
                  }
                }
                result={
                  Object {
                    "alternateNames": Array [
                      "Freddy",
                    ],
                    "fullNames": Array [
                      "Fred Fredricks",
                    ],
                    "localPhotoUrl": null,
                    "personId": "123",
                    "timestamp": "2019-05-15T17:12:23.936282Z",
                    "timestampType": "creation",
                  }
                }
              >
                <SearchResultImpl
                  history={
                    Object {
                      "action": "POP",
                      "block": [Function],
                      "canGo": [Function],
                      "createHref": [Function],
                      "entries": Array [
                        Object {
                          "hash": "",
                          "key": "abc123",
                          "pathname": "/albany/results",
                          "search": "",
                        },
                      ],
                      "go": [Function],
                      "goBack": [Function],
                      "goForward": [Function],
                      "index": 0,
                      "length": 1,
                      "listen": [Function],
                      "location": Object {
                        "hash": "",
                        "key": "abc123",
                        "pathname": "/albany/results",
                        "search": "",
                      },
                      "push": [Function],
                      "replace": [Function],
                    }
                  }
                  intl={
                    Object {
                      "defaultFormats": Object {},
                      "defaultLocale": "en",
                      "formatDate": [Function],
                      "formatHTMLMessage": [Function],
                      "formatMessage": [Function],
                      "formatNumber": [Function],
                      "formatPlural": [Function],
                      "formatRelative": [Function],
                      "formatTime": [Function],
                      "formats": Object {},
                      "formatters": Object {
                        "getDateTimeFormat": [Function],
                        "getMessageFormat": [Function],
                        "getNumberFormat": [Function],
                        "getPluralFormat": [Function],
                        "getRelativeFormat": [Function],
                      },
                      "locale": "en",
                      "messages": Object {},
                      "now": [Function],
                      "onError": [Function],
                      "textComponent": "span",
                      "timeZone": null,
                    }
                  }
                  location={
                    Object {
                      "hash": "",
                      "key": "abc123",
                      "pathname": "/albany/results",
                      "search": "",
                    }
                  }
                  match={
                    Object {
                      "isExact": false,
                      "params": Object {},
                      "path": "/",
                      "url": "/",
                    }
                  }
                  repo={
                    Object {
                      "recordCount": 100,
                      "repoId": "albany",
                      "title": "Albany",
                    }
                  }
                  result={
                    Object {
                      "alternateNames": Array [
                        "Freddy",
                      ],
                      "fullNames": Array [
                        "Fred Fredricks",
                      ],
                      "localPhotoUrl": null,
                      "personId": "123",
                      "timestamp": "2019-05-15T17:12:23.936282Z",
                      "timestampType": "creation",
                    }
                  }
                >
                  <li
                    className="results-result"
                    onClick={[Function]}
                  >
                    <div
                      className="results-resultphoto"
                    >
                      <img
                        src={null}
                      />
                    </div>
                    <div
                      className="results-resultcontent"
                    >
                      <h5
                        className="mdc-typography--headline5"
                      >
                        Fred Fredricks (Freddy)
                      </h5>
                      <p
                        className="mdc-typography--body1"
                      >
                        <FormattedMessage
                          defaultMessage="Record created on {timestampStr}."
                          id="Results.timestampCreation"
                          values={
                            Object {
                              "timestampStr": <FormattedDate
                                day="numeric"
                                hour="numeric"
                                minute="numeric"
                                month="short"
                                timeZoneName="short"
                                value={2019-05-15T17:12:23.936Z}
                              />,
                            }
                          }
                        >
                          <span>
                            Record created on 
                            <FormattedDate
                              day="numeric"
                              hour="numeric"
                              minute="numeric"
                              month="short"
                              timeZoneName="short"
                              value={2019-05-15T17:12:23.936Z}
                            >
                              <span>
                                May 15, 5:12 PM UTC
                              </span>
                            </FormattedDate>
                            .
                          </span>
                        </FormattedMessage>
                      </p>
                    </div>
                  </li>
                </SearchResultImpl>
              </InjectIntl(SearchResultImpl)>
            </Route>
          </withRouter(InjectIntl(SearchResultImpl))>
          <withRouter(InjectIntl(SearchResultImpl))
            key="321"
            repo={
              Object {
                "recordCount": 100,
                "repoId": "albany",
                "title": "Albany",
              }
            }
            result={
              Object {
                "alternateNames": Array [],
                "fullNames": Array [
                  "Alan Smith",
                  "Alan Herbert Smith",
                ],
                "localPhotoUrl": "http://www.example.com/notevenreallylocal.jpg",
                "personId": "321",
                "timestamp": "2019-05-16T16:35:23.936282Z",
                "timestampType": "update",
              }
            }
          >
            <Route>
              <InjectIntl(SearchResultImpl)
                history={
                  Object {
                    "action": "POP",
                    "block": [Function],
                    "canGo": [Function],
                    "createHref": [Function],
                    "entries": Array [
                      Object {
                        "hash": "",
                        "key": "abc123",
                        "pathname": "/albany/results",
                        "search": "",
                      },
                    ],
                    "go": [Function],
                    "goBack": [Function],
                    "goForward": [Function],
                    "index": 0,
                    "length": 1,
                    "listen": [Function],
                    "location": Object {
                      "hash": "",
                      "key": "abc123",
                      "pathname": "/albany/results",
                      "search": "",
                    },
                    "push": [Function],
                    "replace": [Function],
                  }
                }
                location={
                  Object {
                    "hash": "",
                    "key": "abc123",
                    "pathname": "/albany/results",
                    "search": "",
                  }
                }
                match={
                  Object {
                    "isExact": false,
                    "params": Object {},
                    "path": "/",
                    "url": "/",
                  }
                }
                repo={
                  Object {
                    "recordCount": 100,
                    "repoId": "albany",
                    "title": "Albany",
                  }
                }
                result={
                  Object {
                    "alternateNames": Array [],
                    "fullNames": Array [
                      "Alan Smith",
                      "Alan Herbert Smith",
                    ],
                    "localPhotoUrl": "http://www.example.com/notevenreallylocal.jpg",
                    "personId": "321",
                    "timestamp": "2019-05-16T16:35:23.936282Z",
                    "timestampType": "update",
                  }
                }
              >
                <SearchResultImpl
                  history={
                    Object {
                      "action": "POP",
                      "block": [Function],
                      "canGo": [Function],
                      "createHref": [Function],
                      "entries": Array [
                        Object {
                          "hash": "",
                          "key": "abc123",
                          "pathname": "/albany/results",
                          "search": "",
                        },
                      ],
                      "go": [Function],
                      "goBack": [Function],
                      "goForward": [Function],
                      "index": 0,
                      "length": 1,
                      "listen": [Function],
                      "location": Object {
                        "hash": "",
                        "key": "abc123",
                        "pathname": "/albany/results",
                        "search": "",
                      },
                      "push": [Function],
                      "replace": [Function],
                    }
                  }
                  intl={
                    Object {
                      "defaultFormats": Object {},
                      "defaultLocale": "en",
                      "formatDate": [Function],
                      "formatHTMLMessage": [Function],
                      "formatMessage": [Function],
                      "formatNumber": [Function],
                      "formatPlural": [Function],
                      "formatRelative": [Function],
                      "formatTime": [Function],
                      "formats": Object {},
                      "formatters": Object {
                        "getDateTimeFormat": [Function],
                        "getMessageFormat": [Function],
                        "getNumberFormat": [Function],
                        "getPluralFormat": [Function],
                        "getRelativeFormat": [Function],
                      },
                      "locale": "en",
                      "messages": Object {},
                      "now": [Function],
                      "onError": [Function],
                      "textComponent": "span",
                      "timeZone": null,
                    }
                  }
                  location={
                    Object {
                      "hash": "",
                      "key": "abc123",
                      "pathname": "/albany/results",
                      "search": "",
                    }
                  }
                  match={
                    Object {
                      "isExact": false,
                      "params": Object {},
                      "path": "/",
                      "url": "/",
                    }
                  }
                  repo={
                    Object {
                      "recordCount": 100,
                      "repoId": "albany",
                      "title": "Albany",
                    }
                  }
                  result={
                    Object {
                      "alternateNames": Array [],
                      "fullNames": Array [
                        "Alan Smith",
                        "Alan Herbert Smith",
                      ],
                      "localPhotoUrl": "http://www.example.com/notevenreallylocal.jpg",
                      "personId": "321",
                      "timestamp": "2019-05-16T16:35:23.936282Z",
                      "timestampType": "update",
                    }
                  }
                >
                  <li
                    className="results-result"
                    onClick={[Function]}
                  >
                    <div
                      className="results-resultphoto"
                    >
                      <img
                        src="http://www.example.com/notevenreallylocal.jpg"
                      />
                    </div>
                    <div
                      className="results-resultcontent"
                    >
                      <h5
                        className="mdc-typography--headline5"
                      >
                        Alan Smith, Alan Herbert Smith
                      </h5>
                      <p
                        className="mdc-typography--body1"
                      >
                        <FormattedMessage
                          defaultMessage="Record updated on {timestampStr}."
                          id="Results.timestampUpdate"
                          values={
                            Object {
                              "timestampStr": <FormattedDate
                                day="numeric"
                                hour="numeric"
                                minute="numeric"
                                month="short"
                                timeZoneName="short"
                                value={2019-05-16T16:35:23.936Z}
                              />,
                            }
                          }
                        >
                          <span>
                            Record updated on 
                            <FormattedDate
                              day="numeric"
                              hour="numeric"
                              minute="numeric"
                              month="short"
                              timeZoneName="short"
                              value={2019-05-16T16:35:23.936Z}
                            >
                              <span>
                                May 16, 4:35 PM UTC
                              </span>
                            </FormattedDate>
                            .
                          </span>
                        </FormattedMessage>
                      </p>
                    </div>
                  </li>
                </SearchResultImpl>
              </InjectIntl(SearchResultImpl)>
            </Route>
          </withRouter(InjectIntl(SearchResultImpl))>
        </ul>
        <InjectIntl(Footer)>
          <Footer
            intl={
              Object {
                "defaultFormats": Object {},
                "defaultLocale": "en",
                "formatDate": [Function],
                "formatHTMLMessage": [Function],
                "formatMessage": [Function],
                "formatNumber": [Function],
                "formatPlural": [Function],
                "formatRelative": [Function],
                "formatTime": [Function],
                "formats": Object {},
                "formatters": Object {
                  "getDateTimeFormat": [Function],
                  "getMessageFormat": [Function],
                  "getNumberFormat": [Function],
                  "getPluralFormat": [Function],
                  "getRelativeFormat": [Function],
                },
                "locale": "en",
                "messages": Object {},
                "now": [Function],
                "onError": [Function],
                "textComponent": "span",
                "timeZone": null,
              }
            }
          >
            <div
              className="footer"
            >
              <p
                className="mdc-typography--body1"
              >
                <FormattedMessage
                  defaultMessage="PLEASE NOTE: All data entered is available to the public and usable by anyone. Google does not review or verify the accuracy of this data. Google may share the data with public and private organizations participating in disaster response efforts."
                  id="Footer.disclaimerText"
                  values={Object {}}
                >
                  <span>
                    PLEASE NOTE: All data entered is available to the public and usable by anyone. Google does not review or verify the accuracy of this data. Google may share the data with public and private organizations participating in disaster response efforts.
                  </span>
                </FormattedMessage>
              </p>
            </div>
          </Footer>
        </InjectIntl(Footer)>
      </div>
      <RippledComponent
        className="results-addpersonfab"
        disabled={false}
        icon={
          <img
            src="/static/icons/maticon_add.svg"
          />
        }
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        style={Object {}}
        textLabel="Create new record"
        unbounded={false}
      >
        <Component
          className="results-addpersonfab"
          disabled={false}
          icon={
            <img
              src="/static/icons/maticon_add.svg"
            />
          }
          initRipple={[Function]}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          style={Object {}}
          textLabel="Create new record"
        >
          <button
            className="mdc-fab results-addpersonfab mdc-fab--extended"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            style={Object {}}
          >
            <Icon
              icon={
                <img
                  src="/static/icons/maticon_add.svg"
                />
              }
            >
              <img
                className="mdc-fab__icon"
                src="/static/icons/maticon_add.svg"
              />
            </Icon>
            <TextLabel
              textLabel="Create new record"
            >
              <span
                className="mdc-fab__label"
              >
                Create new record
              </span>
            </TextLabel>
          </button>
        </Component>
      </RippledComponent>
    </div>
  </Results>
</InjectIntl(Results)>
`;
